jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -1369,12 +1367,42 @@ public DynamicCapacity()
             this(null, false, -1, -1, -1);
         }
 
+        /**
+         * @param sizedPool The pool from which to allocate buffers, with {@link ByteBufferPool.Sized#isDirect()} configured
+         *                  and {@link ByteBufferPool.Sized#getSize()} used for the size of aggregation buffers.
+         */
+        public DynamicCapacity(ByteBufferPool.Sized sizedPool)
+        {
+            this(null, sizedPool, -1, -1);
+        }
+
+        /**
+         * @param sizedPool The pool from which to allocate buffers, with {@link ByteBufferPool.Sized#isDirect()} configured
+         *                  and {@link ByteBufferPool.Sized#getSize()} used for the size of aggregation buffers.
+         * @param maxSize The maximum length of the accumulated buffers or -1 for 2GB limit
+         */
+        public DynamicCapacity(ByteBufferPool.Sized sizedPool, long maxSize)
+        {
+            this(null, sizedPool, maxSize, -1);
+        }
+
+        /**
+         * @param sizedPool The pool from which to allocate buffers, with {@link ByteBufferPool.Sized#isDirect()} configured
+         *                  and {@link ByteBufferPool.Sized#getSize()} used for the size of aggregation buffers.
+         * @param maxSize The maximum length of the accumulated buffers or -1 for 2GB limit
+         * @param minRetainSize The minimal size of a {@link RetainableByteBuffer} before it will be retained; or 0 to always retain; or -1 for a heuristic;
+         */
+        public DynamicCapacity(ByteBufferPool.Sized sizedPool, long maxSize, int minRetainSize)
+        {
+            this(null, sizedPool, maxSize, minRetainSize);
+        }
+
         /**
          * @param pool The pool from which to allocate buffers
          */
         public DynamicCapacity(ByteBufferPool pool)
         {
-            this(pool, false, -1, -1, -1);
+            this(null, pool instanceof ByteBufferPool.Sized sized ? sized : new ByteBufferPool.Sized(pool), -1, -1);
```

Comment:
I would change this constructor to also take `boolean direct, long maxSize`, this would make it obvious this class wants a size and directness.

Reply:
We have that constructor below.  This is for somebody that has a ByteBufferPool and wants a DynamicCapacity buffer, but is OK with all the defaults.